core/src/test/java/org/opensearch/sql/planner/physical/NestedOperatorTest.java

```diff
@@ -0,0 +1,328 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.opensearch.sql.data.model.ExprValueUtils.collectionValue;
+import static org.opensearch.sql.data.model.ExprValueUtils.tupleValue;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.sql.data.model.ExprNullValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.expression.ReferenceExpression;
+
+@ExtendWith(MockitoExtension.class)
+class NestedOperatorTest extends PhysicalPlanTestBase {
+  @Mock
+  private PhysicalPlan inputPlan;
+
+  private final ExprValue testData = tupleValue(
+      Map.of(
+          "message",
+          collectionValue(
+              List.of(
+                  Map.of("info", "a"),
+                  Map.of("info", "b"),
+                  Map.of("info", "c")
+              )
+          ),
+          "comment",
+          collectionValue(
+              List.of(
+                  Map.of("data", "1"),
+                  Map.of("data", "2"),
+                  Map.of("data", "3")
+              )
+          )
+      )
+  );
+
+
+  private final ExprValue testDataWithSamePath = tupleValue(
+      Map.of(
+          "message",
+          collectionValue(
+              List.of(
+                  Map.of("info", "a", "id", "1"),
+                  Map.of("info", "b", "id", "2"),
+                  Map.of("info", "c", "id", "3")
+              )
+          )
+      )
+  );
+
+  private final ExprValue nonNestedTestData = tupleValue(
+      Map.of(
+          "message", "val"
+      )
+  );
+
+  private final ExprValue missingArrayData = tupleValue(
+      Map.of(
+          "missing",
+          collectionValue(
+              List.of("value")
+          )
+      )
+  );
+
+  @Test
+  public void nested_one_nested_field() {
+    when(inputPlan.hasNext()).thenReturn(true, false);
+    when(inputPlan.next())
+        .thenReturn(testData);
+
+    Set<String> fields = Set.of("message.info");
+    Map<String, List<String>> groupedFieldsByPath =
+        Map.of("message", List.of("message.info"));
+
+    assertThat(
+        execute(new NestedOperator(inputPlan, fields, groupedFieldsByPath)),
+        contains(
+            tupleValue(
+                new LinkedHashMap<>() {{
+                  put("message.info", "a");
+                  put("comment", collectionValue(
+                      new ArrayList<>() {{
+                        add(new LinkedHashMap<>() {{
+                            put("data", "1");
```

Comment:
Maybe (optional): `List.of(new LinkedHashMap<>(Map.of(...)), new LinkedHashMap<>(Map.of(...)), ...)`